WordPress/Classes/Models/Blog.m

```diff
@@ -235,25 +238,55 @@ - (NSArray *)sortedCategories
     return [[self.categories allObjects] sortedArrayUsingDescriptors:sortDescriptors];
 }
 
+- (NSArray *)sortedPostFormats
+{
+    if ([self.postFormats count] == 0) {
+        return @[];
+    }
+    NSMutableArray *sortedFormats = [NSMutableArray arrayWithCapacity:[self.postFormats count]];
+ 
+    if (self.postFormats[@"standard"]) {
+        [sortedFormats addObject:@"standard"];
+    }
+    [self.postFormats enumerateKeysAndObjectsUsingBlock:^(id key, id obj, BOOL *stop) {
+        if (![key isEqual:@"standard"]) {
+            [sortedFormats addObject:key];
+        }
+    }];
+
+    return [NSArray arrayWithArray:sortedFormats];
+}
+
 - (NSArray *)sortedPostFormatNames
 {
     NSMutableArray *sortedNames = [NSMutableArray arrayWithCapacity:[self.postFormats count]];
 
-    if ([self.postFormats count] != 0) {
-        id standardPostFormat = [self.postFormats objectForKey:@"standard"];
-        if (standardPostFormat) {
-            [sortedNames addObject:standardPostFormat];
-        }
-        [self.postFormats enumerateKeysAndObjectsUsingBlock:^(id key, id obj, BOOL *stop) {
-            if (![key isEqual:@"standard"]) {
-                [sortedNames addObject:obj];
-            }
-        }];
+    for (NSString *key in self.sortedPostFormats) {
+        [sortedNames addObject:self.postFormats[key]];
```

Comment:
> I see that self hosted sites don't sync the values with the server, but still apply them locally, which seems good enough to me. Is that the desired behavior?

Yep, because I think it avoids unnecessary work for users in the mobile app that create new posts and always needed to changes this.